tests/controller_tests/test_filters.py

```diff
@@ -76,3 +82,60 @@ def test_get_outcome_categories(seeded_app_ctx):
 
     assert response_json
     assert all(isinstance(cat, str) for cat in response_json)
+
+
+def test_get_reporting_period_range_not_found(app: FlaskClient):
+    """Asserts failed retrieval of funds."""
+
+    response = app.get("/reporting-period-range")
+
+    assert response.status_code == 404
+    assert response.json["detail"] == "No reporting period range found."
+
+
+def test_get_reporting_period_range(app_ctx):
+    """Asserts successful retrieval of financial periods."""
+
+    # Create some sample submissions for testing
+    sub1 = Submission(
+        submission_id="1",
+        submission_date=datetime(2023, 5, 1),
+        reporting_period_start=datetime(2023, 4, 1),
+        reporting_period_end=datetime(2023, 4, 30),
+        reporting_round=1,
+    )
+    sub2 = Submission(
+        submission_id="2",
+        submission_date=datetime(2024, 5, 5),
+        reporting_period_start=datetime(2024, 5, 1),
+        reporting_period_end=datetime(2024, 5, 31),
+        reporting_round=2,
+    )
+    sub3 = Submission(
+        submission_id="3",
+        submission_date=datetime(2025, 6, 1),
+        reporting_period_start=datetime(2025, 6, 1),
+        reporting_period_end=datetime(2025, 6, 30),
+        reporting_round=1,
+    )
+    sub4 = Submission(
+        submission_id="4",
+        submission_date=datetime(2021, 6, 5),
+        reporting_period_start=datetime(2021, 6, 1),
+        reporting_period_end=datetime(2021, 6, 30),
+        reporting_round=2,
+    )
+    submissions = [sub1, sub2, sub3, sub4]
+    db.session.add_all(submissions)
+    db.session.flush()
```

Comment:
we actually don't need the db.session.flush():

This actually is to run cascade deletions (ie flush out all the unwanted records). I think there are a few of these, they are a legacy of some copy-and-paste test code. We should remove all these others in a separate PR I guess